src/random.cpp

```diff
@@ -326,6 +359,19 @@ static void SeedSlow(CSHA512& hasher)
 
     // Windows performance monitor data (once every 10 minutes)
     RandAddSeedPerfmon(hasher);
+
+    // Once per minute, strengthen the hash for 10ms.
+    static std::atomic<int64_t> last_strengthen;
+    int64_t current_time = GetTimeMicros();
+    if (current_time > last_strengthen + 60000) {
+        // Write randomness from the actual RNG in the state, so it goes through the strengthening as well.
+        unsigned char buffer[32];
+        GetRandBytes(buffer, 32);
```

Comment:
I'd kinda feel better about this if it just directly took the cs_rng_state lock and read the state here instead of calling GetRandBytes... at least then the code won't mysteriously crash if someone upgrades GetRandBytes to use ::Slow.

Reply:
I've abstracted the update-the-global-RNG-and-extract-entropy into a separate function, which is invoked from both this seeder and RandProc.